Mapping Tools/Classes/SystemTools/ListenerManager.cs

```diff
@@ -104,6 +104,52 @@ public class ListenerManager {
             EditorReaderStuff.CoolSave();
         }
 
+        private void SmartQuickRun() {
+            if (!SettingsManager.Settings.SmartQuickRunEnabled) { QuickRunCurrentTool(); return; }
+
+            if (EditorReaderStuff.TryGetFullEditorReader(out EditorReader reader)) {
+                EditorReaderStuff.GetNewestVersion(IOHelper.GetCurrentBeatmap(), out List<BeatmapHelper.HitObject> selected, reader);
+                int so = selected.Count;
+
+                IQuickRun tool = null;
+
+                System.Windows.Application.Current.Dispatcher.Invoke(() => {
+                    if (so <= 1) {
+                        switch (SettingsManager.Settings.SingleQuickRunTool) {
+                            case SingleQuickRunEnum.Current:
```

Comment:
you could check if all settings are set to current to do QuickRunCurrentTool before using editor reader.